core/src/main/java/org/elasticsearch/Version.java

```diff
@@ -55,33 +55,25 @@
     public static final Version V_5_0_1 = new Version(V_5_0_1_ID, org.apache.lucene.util.Version.LUCENE_6_2_1);
     public static final int V_5_0_2_ID = 5000299;
     public static final Version V_5_0_2 = new Version(V_5_0_2_ID, org.apache.lucene.util.Version.LUCENE_6_2_1);
-    public static final int V_5_0_3_ID_UNRELEASED = 5000399;
```

Comment:
I think that we can enforce that these are marked as released in the same way that we enforce the list of upgrade versions in the packaging tests is bumped any time there is a release.